blocks/api/factory.js

```diff
@@ -85,84 +87,143 @@ export function cloneBlock( block, mergeAttributes = {}, newInnerBlocks ) {
 }
 
 /**
- * Returns a predicate that receives a transformation and returns true if the
- * given transformation is able to execute in the situation specified in the
- * params.
+ * Returns a boolean indicating whether a transform is possible based on
+ * various bits of context
  *
- * @param {string}  sourceName   Block name.
- * @param {boolean} isMultiBlock Array of possible block transformations.
+ * @param {Object} transform The transform object to validate
+ * @param {string} direction Is this a 'from' or 'to' transform
+ * @param {Array} blocks The blocks to transform from
+ * @param {boolean} isMultiBlock Have multiple blocks been selected?
  *
- * @return {Function} Predicate that receives a block type.
+ * @return {boolean} Is the transform possible?
  */
-const isTransformForBlockSource = ( sourceName, isMultiBlock = false ) => ( transform ) => (
-	transform.type === 'block' &&
-	transform.blocks.indexOf( sourceName ) !== -1 &&
-	( ! isMultiBlock || transform.isMultiBlock )
-);
+const isPossibleTransformForSource = ( transform, direction, blocks, isMultiBlock ) => {
+	if ( isEmpty( blocks ) ) {
+		return false;
+	}
+	const sourceBlock = first( blocks );
+
+	// If multiple blocks are selected, only multi block transforms are allowed
+	const isValidForMultiBlocks = ! isMultiBlock || transform.isMultiBlock;
+	if ( ! isValidForMultiBlocks ) {
+		return false;
+	}
+
+	// Only consider 'block' type transforms as valid
+	const isBlockType = transform.type === 'block';
+	if ( ! isBlockType ) {
+		return false;
+	}
+
+	// Check if the transform's block name matches the source block only if this is a transform 'from'
+	const hasMatchingName = direction !== 'from' || transform.blocks.indexOf( sourceBlock.name ) !== -1;
+	if ( ! hasMatchingName ) {
+		return false;
+	}
+
+	// If the transform has a `canTransform` function specified, check that it returns true
+	if ( isFunction( transform.canTransform ) ) {
+		const attributes = transform.isMultiBlock ? blocks.map( ( block ) => block.attributes ) : sourceBlock.attributes;
+		if ( ! transform.canTransform( attributes ) ) {
```

Comment:
Any thoughts on whether to name this callback `isMatch`? This would align with the `isMatch` callbacks we have on `raw` and `file` transformations. Consistency might help plugin developers discover/learn the API.

Reply:
Initially in my head it was something different - given the params are different and it's implemented in a different place. 

Looking at it again though I think you're right, it should be `isMatch` given it does almost precisely the same thing :)